### PR TITLE
[PXM-41] Added reserved gods minting

### PR DESCRIPTION
**Pull request by GabrielCartier**

### CODE REVIEW

File: contracts/Genesis.sol
Comment (maoueh) [inline contracts/Genesis.sol:183]: Each `GenesisSupply(genesisSupplyAddress)` increased the gas cost by some good factor, making a call cost minimally 2100 gas.

Couldn't just we forward everything to the supply contract?

At the very least, some refactoring should be done to go from 3 calls to 1 that returns max supply + actual index.

File: contracts/GenesisSupply.sol
Comment (maoueh) [inline contracts/GenesisSupply.sol:189]: Shouldn't `tokenCounter` be incremented as well?

File: contracts/GenesisSupply.sol
Comment (maoueh) [inline contracts/GenesisSupply.sol:256]: Micro-optimizations, I would put `isRevealed` condition first so most call cost barely nothing to run, only when not reveal that extra gas will need to be paid.

**Review by maoueh — APPROVED** [review]
Approving so we can move forward, but check my comments

File: contracts/Genesis.sol
Comment (GabrielCartier) [reply contracts/Genesis.sol:183]: Ok, I'll check this after the merge, good call, thx for pointing it out!

File: contracts/GenesisSupply.sol
Comment (GabrielCartier) [reply contracts/GenesisSupply.sol:189]: no because in the constructor we already increment this to 10 initially. Kinda shitty design IMO, but we save on gas on construction when counters start at something other than 0. The comment explains it, but yea, not super clear.

File: contracts/GenesisSupply.sol
Comment (GabrielCartier) [reply contracts/GenesisSupply.sol:256]: Can't do that because backend has access to this BEFORE the reveal date to be able to generate the NFTs before the reveal.

**Pull request merged**
